### ISSUE TITLE
connection timeout when using TLS on Redis Cluster

### ISSUE THREAD

**Issue by dustinrue** [predis/predis#1474]
**Describe the bug**
I am running AWS ElastiCache (Redis) in multi-az cluster mode with TLS enabled and required. In this mode, connections to Redis timeout because predis is not using tls but is instead hard coded to use tcp (https://github.com/predis/predis/blob/v2.x/src/Connection/StreamConnection.php#L133). Changing this from tcp to tls allows connections to work as expected.

**To Reproduce**
1. Configure AWS ElastiCache in redis 7 cluster mode
2. Enforce TLS connections (TLS required mode)
3. Configure WordPress with the redis-cache plugin (configuration shown below)
4. Attempt to do anything, Redis connections will timeout

```
define( 'WP_REDIS_HOST', 'clustercfg.redis-cluster-test.[redacted].use1.cache.amazonaws.com' );
define( 'WP_REDIS_PORT', '6379' );

define( 'WP_REDIS_PASSWORD', '[redacted]' );
define( 'WP_REDIS_TIMEOUT', 1 );
define( 'WP_REDIS_READ_TIMEOUT', 1 );
define( 'WP_REDIS_DATABASE', 0 );

define( 'WP_REDIS_SCHEME', 'tls' );
define( 'WP_REDIS_CLIENT', 'predis' );
define( 'WP_REDIS_CLUSTER', [
  'tcp://redis-cluster-test-0001-001.redis-cluster-test.[redacted].use1.cache.amazonaws.com:6379?alias=node-01',
  'tcp://redis-cluster-test-0001-002.redis-cluster-test.[redacted].use1.cache.amazonaws.com:6379?alias=node-02',
] );
```

Note that changing how cluster members are defined from tcp to tls doesn't affect anything. The scheme is effectively dropped in this routine - https://github.com/rhubarbgroup/redis-cache/blob/develop/includes/object-cache.php#L3027. predis appears to only be made aware of the need for TLS because WP_REDIS_SCHEME is set to `tls` or `rediss` and this will be sent as a parameter value.

**Expected behavior**
Expectation is that TLS connections work properly

**Versions (please complete the following information):**
- Predis: 2.1.2 but is present in the code in the main branch yet
- PHP 8.3.10
- Redis Server [e.g. 7]
- OS [e.g. Ubuntu 20.10]

**Code sample**
See issue description

**Additional context**
I only went as far as changing the hardcoded value from tcp to tls and did not dig deeper. Once I changed it to tls the Redis connection started working as expected and I did not dig further. There may be routines that modify the scheme elsewhere in the code that is not being triggered due to some other configuration error on my part. If it is a configuration error then this is a documentation issue as I can't find any mention about TLS other than to set WP_REDIS_SCHEME to tls.

**Comment by tillkruss** [comment]
The `WP_REDIS_SCHEME` is ignored, since you're setting `tcp://` in `WP_REDIS_CLUSTER`.

**Comment by dustinrue** [comment]
Yes, that seemed like an obvious thing to do which I did try before submitting the issue. Unfortunately, the connection still doesn't work properly. Even if the connection is properly initialized in https://github.com/rhubarbgroup/redis-cache/blob/04176f4c0d9c39eaa560768ef0730f2b69de3ba2/dependencies/predis/predis/src/Connection/StreamConnection.php#L194 and it immediately calls https://github.com/rhubarbgroup/redis-cache/blob/04176f4c0d9c39eaa560768ef0730f2b69de3ba2/dependencies/predis/predis/src/Connection/StreamConnection.php#L130 which hard codes the address to use tcp://. Changing WP_REDIS_CLUSTER to the corrected scheme is never seen or dealt with in the predis code as far as I can tell.

In other words, despite having

```
define( 'WP_REDIS_CLUSTER', [
  'tls://redis-cluster-test-0001-001.redis-cluster-test.[redacted].use1.cache.amazonaws.com:6379?alias=node-01',
  'tls://redis-cluster-test-0001-002.redis-cluster-test.[redacted].use1.cache.amazonaws.com:6379?alias=node-02',
] );
```

in my wp-config.php

I still see this error:

```
wp plugin list
Predis\Connection\ConnectionException: Error while reading line from the server. [tcp://redis-cluster-test-0001-001.redis-cluster-test.[redacted].use1.cache.amazonaws.com:6379] in /var/www/html/wordpress/wp-content/plugins/redis-cache/dependencies/predis/predis/src/Connection/AbstractConnection.php:131
```

**Comment by tillkruss** [comment]
@vladvildanov Have you tried this?
